docs/developer-guide/extensions/proxy-extensions.md

```diff
@@ -256,6 +286,27 @@ is then sanitized before being sent to the backend service. The
 request sanitization will remove sensitive information from the
 request like the `Cookie` and `Authorization` headers.
 
+A new `Authorization` header can be added to the outgoing request by
+defining it as a header in the `extensions.backend.services.headers`
+configuration. Consider the following example:
+
+```yaml
+extension.config: |
+  extensions:
+  - name: some-extension
+    backend:
+      services:
+      - url: http://extension-name.com:8080
+        headers:
+        - name: Authorization
+          valueSecretRef: '$some-extension.authorization.header'
+```
+
+In the example above, all requests sent to
+`http://extension-name.com:8080` will have an additional
+`Authorization` header. The value of this header will be the one from
+the `argocd-secret` with key `some-extension.authorization.header`
```

Comment:
When we add TLS support, this would be a good place to mention that, if you're sending auth info, it's probably smart to use TLS.